addons/mrp/tests/test_bom.py

```diff
@@ -2443,6 +2444,54 @@ def test_archive_operation(self):
         self.assertEqual(bom.bom_line_ids[1].operation_id, ope_2)
         self.assertEqual(bom.byproduct_ids[1].operation_id, ope_2)
 
+    def add_variant_to_product(self, product):
```

Comment:
The name is a bit confusing and the method actually expects a template
```suggestion
    def add_attribute_to_template(self, template):
```
Also, we try to always gather helpers at the beginning of the class